src/FileSystem-Core/MemoryDirectoryEntry.class.st

```diff
@@ -70,3 +70,10 @@ MemoryDirectoryEntry >> posixPermissions [
 	
 	^8r777
 ]
+
+{ #category : #accessing }
+MemoryDirectoryEntry >> size [ 
+	^self isFile 
```

Comment:
Thanks again for submitting the fix.

MemoryFileSystemDirectory knows its size (0), so there's no need to have the test in MemoryDirectoryEntry>> size.  It can be just ^reference size.